exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.cpp

```diff
@@ -0,0 +1,135 @@
+#include "wrapper/algo_wrapper/binary/binary_wrapper.hpp"
+
+#include "wrapper/algo_wrapper/wrapper.hpp"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <fmt/core.h>
+#include <sys/mman.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <cerrno>
+#include <cstdio>
+#include <cstring>
+
+#include <system_error>
+
+namespace nutc {
+namespace wrapper {
+
+BinaryWrapper::BinaryWrapper(
+    std::string algo, std::string trader_id, MarketOrderFunc publish_market_order
+) : Wrapper(std::move(algo), std::move(trader_id))
+{
+    // Create a memory file descriptor
+    memfd_ = memfd_create("wrapper-library", MFD_CLOEXEC);
+    if (memfd_ == -1) {
+        throw std::system_error(errno, std::generic_category(), "memfd_create failed");
+    }
+
+    auto bytes_written = write(memfd_, algo_.data(), algo_.size());
+    if (bytes_written == -1 || static_cast<size_t>(bytes_written) != algo_.size()) {
+        close(memfd_);
```

Comment:
There are better ways to do this than having close and dlclose for every bad condition

Reply:
I'd support try/catch instead but up to you

Reply:
I think smart pointers have a way to let you use RAII here.. need to look into it again though cause I forgot exactly how it works

Reply:
You're writing c-style code, that means you need need to do c-style exception handling. On error, you need to use a `goto` to jump into the cleanup block at the end of your function.